tsdb/tx.go

```diff
@@ -316,7 +316,14 @@ func (l *LocalMapper) Begin(c *influxql.Call, startingTime int64, chunkSize int)
 	if fieldName != "" {
 		fid, err := l.decoder.FieldIDByName(fieldName)
 		if err != nil {
-			return fmt.Errorf("%s isn't a field on measurement %s", fieldName, l.job.MeasurementName)
+			switch {
+			case c.Name == "distinct":
```

Comment:
Is this really necessary? Why richer messages fro `distinct` and `count distinct`?

I mean, I like helpful messages for the user, I'm just not sure why these are special-cased.